misk/src/main/kotlin/misk/web/Request.kt

```diff
@@ -3,13 +3,69 @@ package misk.web
 import misk.web.actions.WebSocket
 import okhttp3.Headers
 import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.RequestBody
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.BufferedSink
 import okio.BufferedSource
 import org.eclipse.jetty.http.HttpMethod
+import java.io.IOException
+import java.net.HttpURLConnection
 
 data class Request(
   val url: HttpUrl,
-  val method: HttpMethod,
+  val method: HttpMethod = HttpMethod.GET,
   val headers: Headers = Headers.of(),
   val body: BufferedSource,
   val websocket: WebSocket? = null
 )
+
+fun okhttp3.Request.exec(): okhttp3.Response {
+  val client = OkHttpClient()
+  return try {
+    client.newCall(this).execute()
+  } catch (e: IOException) {
+    Response.Builder()
+        .request(this)
+        .protocol(Protocol.HTTP_1_1)
+        .addHeader("Content-Type", "text/plain; charset=utf-8")
+        .body(ResponseBody.create(MediaType.parse("Failed to fetch upstream URL ${this.url()}"), "Failed to fetch upstream URL ${this.url()}"))
+        .message("Failed to fetch upstream URL ${this.url()}")
+        .code(HttpURLConnection.HTTP_UNAVAILABLE)
+        .build()
+  }
+}
+
+fun Request.forwardRequestTo(proxyUrl: HttpUrl): misk.web.Response<*> {
+  return this.toOkHttp3().setUrl(proxyUrl).exec().toMisk()
```

Comment:
nit: we prefer to omit `this.` prefixes unless they’re required. 

Shipley’s law!
https://twitter.com/jessewilson/status/417478030806622208?lang=en